tests/test_baseline.py

```diff
@@ -0,0 +1,275 @@
+        self.model = load_model("", device=device)
+        self.frames = torch.rand(512, self.model.n_fft, device=device) * 2 - 1
+
+    def teardown(self):
+        del self.model
+        del self.frames
+
+    def test_model_sliced(self):
+        frames_sliced = self.frames[::2, ...]
+        assert torch.all(frames_sliced[0] - self.frames[0] == 0)
+        assert torch.all(frames_sliced[1] - self.frames[2] == 0)
+        assert torch.all(frames_sliced - self.frames[::2, ...] == 0)
+
+        outputs = self.model(self.frames)
+        outputs_sliced = self.model(frames_sliced)
+
+        for output, output_sliced in zip(outputs.values(), outputs_sliced.values()):
+            assert torch.all(torch.abs(output_sliced[0] - output[0]) < 1e-5)
+            assert torch.all(torch.abs(output_sliced[1] - output[2]) < 1e-5)
+            assert torch.all(torch.abs(output_sliced - output[::2, ...]) < 1e-5)
+
+
+class TestLayerbyLayer:
+    def test_layers_find_error(self):
+        device = "cuda:0" if torch.cuda.is_available() else "cpu"
+        model = load_model("", device=device)
+
+        frames = torch.rand(512, model.n_fft, device=device)
+        frames_sliced = frames[::2, ...]
+        assert torch.all(torch.abs(frames[2] - frames_sliced[1]) == 0)
+
+        # Layer by the layer perform the same operation on the sliced and the whole frame.
+        # The current error cap is set by changing layer by layer and setting the max possible error.
+        # The purpose is to understand why the batched tests are failing.
+        x = torch.fft.rfft(frames * model.window)
+        y = torch.fft.rfft(frames_sliced * model.window)
+        assert torch.all(torch.abs(x[::2, ...] - y) < 1e-25)
+
+        x = torch.abs(x) ** 2.0
+        y = torch.abs(y) ** 2.0
+        assert torch.all(torch.abs(x[::2, ...] - y) < 1e-25)
+
+        # The matmul here is the first point where the error increases to 1e-5
+        x = torch.matmul(x, model.mel_scale.transpose(0, 1))
+        y = torch.matmul(y, model.mel_scale.transpose(0, 1))
+        assert torch.all(torch.abs(x[::2, ...] - y) < 1e-5)
+
+        x = torch.log(x + model.epsilon)
+        y = torch.log(y + model.epsilon)
+        assert torch.all(torch.abs(x[::2, ...] - y) < 1e-6)
+
+        # Subsequent increase in error is at the matmuls for the different
+        # embeddings shape.
+        x4096 = x.matmul(model.emb4096)
+        y4096 = y.matmul(model.emb4096)
+        assert torch.all(torch.abs(x4096[::2, ...] - y4096) < 1e-4)
+
+        x2048 = x4096.matmul(model.emb2048)
+        y2048 = y4096.matmul(model.emb2048)
+        assert torch.all(torch.abs(x2048[::2, ...] - y2048) < 1e-3)
+
+        x512 = x2048.matmul(model.emb512)
+        y512 = y2048.matmul(model.emb512)
+        assert torch.all(torch.abs(x512[::2, ...] - y512) < 1e-2)
+
+        x128 = x512.matmul(model.emb128)
+        y128 = y512.matmul(model.emb128)
+        assert torch.all(torch.abs(x128[::2, ...] - y128) < 1e-1)
```

Comment:
I suggested to @jorshi that we not cascade these, and instead project down from x onto each embedding size individually.